app/Models/Device.php

```diff
@@ -141,6 +141,18 @@ public static function findByIp($ip)
 
         return null;
     }
+    
+    public static function findByHostnameOrIp($hostnameOrIp)
```

Comment:
Heh, just took a look at this and this should be the entire function:
```php
public static function findByHostnameOrIp($hostnameOrIp)
{
    return Device::findByIp($hostnameOrIp) ?: Device::findByHostname($hostnameOrIp);
}
```

Reply:
Oh yes, you are right, didn't see that.